web/src/inbox_ui.ts

```diff
@@ -1323,6 +1324,58 @@ export function collapse_or_expand(container_id: string): void {
     }
 
     save_data_to_ls();
+    update_collapsed_note_visibility();
+}
+
+// We show the note "All of your unread conversations are hidden.
+// Click on a section, folder, or channel to see what's inside" for
+// the following situations:
+//   - All folders collapsed.
+//   - If all folders are not collapsed, all visible channels are collapsed.
+// For all other cases, the note is hidden.
+function update_collapsed_note_visibility(): void {
+    const $folders = $("#inbox-list .inbox-folder:not(.hidden_by_filters)");
+    if ($folders.length === 0) {
+        $("#inbox-collapsed-note").hide();
+        return;
+    }
+
+    const any_expanded_folders = $folders.toArray().some((folder) => {
+        const $folder = $(folder);
+        if (!$folder.hasClass("inbox-collapsed-state")) {
+            if (
+                $folder.attr("id") === "inbox-dm-header" &&
+                !$folder.hasClass("hidden_by_filters")
+            ) {
+                return true;
+            }
+            const $channels = $folder
+                .next(".inbox-folder-components")
+                .find(".inbox-header:not(.hidden_by_filters)");
+            return $channels.filter(":not(.inbox-collapsed-state)").length > 0;
+        }
+        return false;
```

Comment:
I would expect this to use data structures like `collapsed_containers` rather than so much JQuery.